docs/sources/flow/reference/components/pyroscope.scrape.md

```diff
@@ -32,45 +55,104 @@ pyroscope.scrape "LABEL" {
-attributes are being passed (e.g., defining both a BearerToken and
-BearerTokenFile or configuring both Basic Authorization and OAuth2 at the same
-time), the component reports an error.
+Any omitted arguments take on their default values. If conflicting
+arguments are being passed (for example, configuring both `bearer_token` 
+and `bearer_token_file`), then `pyroscope.scrape` will fail to start and will report an error.
 
 The following arguments are supported:
 
-Name | Type | Description | Default | Required
----- | ---- | ----------- | ------- | --------
-`targets`                  | `list(map(string))`     | List of targets to scrape. | | yes
-`forward_to`               | `list(ProfilesReceiver)` | List of receivers to send scraped profiles to. | | yes
-`job_name`                 | `string`   | The job name to override the job label with. | component name | no
-`params`                   | `map(list(string))` | A set of query parameters with which the target is scraped. | | no
-`scrape_interval`          | `duration` | How frequently to scrape the targets of this scrape config. | `"15s"` | no
-`scrape_timeout`           | `duration` | The timeout for scraping targets of this config. | `"15s"` | no
-`scheme`                   | `string`   | The URL scheme with which to fetch metrics from targets. | | no
-`bearer_token`             | `secret`   | Bearer token to authenticate with. | | no
-`bearer_token_file`        | `string`   | File containing a bearer token to authenticate with. | | no
-`proxy_url`                | `string`   | HTTP proxy to proxy requests through. | | no
-`follow_redirects`         | `bool`     | Whether redirects returned by the server should be followed. | `true` | no
-`enable_http2`             | `bool`     | Whether HTTP2 is supported for requests. | `true` | no
-
- At most one of the following can be provided:
+Name                | Type                     | Description                                                  | Default        | Required
+------------------- | ------------------------ | ------------------------------------------------------------ | -------------- | --------
+`targets`           | `list(map(string))`      | List of targets to scrape.                                   |                | yes
+`forward_to`        | `list(ProfilesReceiver)` | List of receivers to send scraped profiles to.               |                | yes
+`job_name`          | `string`                 | The job name to override the job label with.                 | component name | no
+`params`            | `map(list(string))`      | A set of query parameters with which the target is scraped.  |                | no
+`scrape_interval`   | `duration`               | How frequently to scrape the targets of this scrape config.  | `"15s"`        | no
+`scrape_timeout`    | `duration`               | The timeout for scraping targets of this config.             | `"18s"`        | no
```

Comment:
I think it's best to add it right here: "It must be larger than `scrape_interval`".

Reply:
I'm not too sure about this one. We usually document constraints under the table... I'll let the comment sit for now in case someone else wants to opine.

Reply:
cc @cyriltovena curious your thoughts here

Reply:
I think it is a bug to require scrape_timeout longer that scrape_interval.
It may make sense for cpu, but does not makes sense for other (not delta profiles) requests which should not last long

Reply:
I see. Shouldn't the docs therefore explain this - that CPU profiles may need different timeout & interval than the remaining profile types?

Reply:
For now I will just state "must be larger than `scrape_interval`" in the table as @thampiotr  suggested. We usually state such things under the table, but I think it improves readability in this case and it's worth making an exception.

> Shouldn't the docs therefore explain this - that CPU profiles may need different timeout & interval than the remaining profile types?

I think for now we can just explain how the component behaves, and we can improve with such advice later on. I think what @korniltsev states applies not just for CPU, but any "delta" profile.

Btw I think the reason for timing out delta profiles before the next query is started is that there can't be more than one active pprof for a certain type at a time. However, even if we have a very slow non-delta pprof query, I suppose we still won't be able to send a new query until the old one has finished. This is just a guess though.